src/pages/dmt5/_margin_calculator.js

```diff
@@ -0,0 +1,292 @@
+    font-size: 14px;
+    height: 4rem;
+    width: 21rem;
+    @media ${device.mobileL} {
+        padding: 1rem 1.5rem;
+        width: 198px;
+    }
+
+    :active {
+        outline: none;
+        border: none;
+    }
+    :focus {
+        outline: 0;
+    }
+`
+
+const StyledFlexContainer = styled(Flex)`
+    border: 1px solid var(--color-grey-33);
+`
+
+const StyledFlexRight = styled(Flex)`
+    margin-right: 2.4rem;
+    @media ${device.mobileL} {
+        margin: 0;
+    }
+`
+
+const StyledFlexLeft = styled(Flex)`
+    background-color: var(--color-grey-25);
+    margin: 0;
+`
+
+const CalculatorCard = ({ name, image, image_alt_name, text, link, button_text, is_mobile }) => (
+    <Flex
+        jc="center"
+        tablet_jc="center"
+        ai="center"
+        fd="column"
+        wrap={is_mobile ? 'nowrap' : 'wrap'}
+        width={is_mobile ? '100%' : '58.8rem'}
+        height={is_mobile ? 'auto' : '69.4rem'}
+    >
+        <SubHeader align="center">{name}</SubHeader>
+        <CardText align="center">{text}</CardText>
+        <ImageWrapper>
+            <QueryImage data={image} alt={image_alt_name} />
+        </ImageWrapper>
+        <StyledLinkButton tertiary="true" to={link} external="true" target="_blank">
+            <Localize translate_text={button_text} />
+        </StyledLinkButton>
+    </Flex>
+)
+
+CalculatorCard.propTypes = {
+    button_text: Proptypes.string,
+    image: Proptypes.object,
+    image_alt_name: Proptypes.string,
+    image_name: Proptypes.string,
+    is_mobile: Proptypes.bool,
+    link: Proptypes.string,
+    name: Proptypes.string,
+    text: Proptypes.string,
+}
+
+const calculators = [
+    {
+        index: 0,
+        name: <Localize translate_text="Margin calculator" />,
+        text: (
+            <Localize translate_text="Calculate the margin you need to open and hold your positions with our margin calculator." />
+        ),
+        image_name: 'margin_calculator',
+        button_text: 'Try our margin calculator',
+        link: '/trader-tools#marginCalculator',
+    },
+    {
+        index: 1,
+        name: <Localize translate_text="Swap calculator" />,
+        text: (
+            <Localize translate_text="Calculate your swap fee and know exactly what you are expected to pay or will earn for maintaining an overnight contract." />
+        ),
+        image_name: 'swap_calculator',
+        button_text: 'Try our swap calculator',
+        link: '/trader-tools/#swapCalculator',
+    },
+]
+
+const MarginCalculator = () => {
+    const [is_mobile, setMobile] = useState(false)
+    const handleResizeWindow = () => {
+        setMobile(isBrowser() ? window.screen.width <= size.tabletS : false)
+    }
+    useEffect(() => {
+        setMobile(isBrowser() ? window.screen.width <= size.tabletS : false)
+        window.addEventListener('resize', handleResizeWindow)
+
+        return () => {
+            window.removeEventListener('resize', handleResizeWindow)
+        }
+    })
```

Comment:
I think we can use the `Show` component for whether it's mobile or desktop. Should be more performant.

Reply:
we have file called os-detect.js to check whether the current view is mobile or desktop logically. you can add this function to enhance the logic there. but preferrably using Show component is better